File: Python_Code/ch3/test1.py
"""
Copyright (c) 2020 WEI.ZHOU. All rights reserved.
The following code snippets are only used for circulation and cannot be used for business.
If the code is used, no consent is required, but the author has nothing to do with any problems and consequences.
In case of code problems, feedback can be made through the following email address.
                                   <xiaoandx@gmail.com>
"""
# 声明保存成绩列表
recordList = []
refereeNumber = int(input("请输入评委人数:"))

# 判断评委人数是否大于2人
while refereeNumber <= 2:
    print("评委人数需大于2")
    refereeNumber = int(input("输入评委人数："))

number = refereeNumber


# 判断输入成绩是否满足0-100
def check_score(score):
    if 0 <= score <= 100:
        return score
    else:
        print("分数不合法，重新输入！")
        return check_score(int(input("输入分数：")))


# 根据评委人数循环输入成绩
while number > 0:
    score = check_score(int(input("输入分数:")))
    recordList.append(score)
    number -= 1

# 计算平均成绩
avg_score = (sum(recordList) - min(recordList) - max(recordList)) / (refereeNumber - 2)

print("选手的平均成绩为 {:.2} 分".format(avg_score))

File: C_Code/basics/chaptersFour2/expOne/test1.c
/*
 * @explain: Copyright (c) 2020 WEI.ZHOU. All rights reserved.
 * The following code is only used for learning and communication, not for
 * illegal and commercial use. If the code is used, no consent is required, but
 * the author has nothing to do with any problems and consequences.
 *
 * In case of code problems, feedback can be made through the following email
 * address. <xiaoandx@gmail.com>
 *
 * @Description:
 * @Author: WEI.ZHOU
 * @Date: 2020-11-07 16:02:37
 * @Version: V1.0
 * @LastEditTime: 2020-11-07 16:04:45
 * @LastEditors: WEI.ZHOU
 * @Others:
 */
#include <stdio.h>
int main() {
    int i, s = 1;
    //因为到第十天还剩一个，所以循环9次
    for (i = 0; i < 9; i++){   
        //倒序求出每天的数量
        s = (s + 1) * 2;  
    }
    //输出结果
    printf("第一天共摘了%d多少桃子\n", s);  
    return 0;
}